modules/mbedtls/configs/config-tls-generic.h

```diff
@@ -241,6 +241,7 @@
 #endif
 
 #if defined(CONFIG_MBEDTLS_MAC_SHA256_ENABLED)
+#define MBEDTLS_SHA224_C
```

Comment:
This configuration enables `MBEDTLS_SHA512_C`. In Mbed TLS 2.x this enabled SHA-384, but in 3.0 this is no longer the case. So you should probably add `#define MBEDTLS_SHA384_C` as well.